server/auth_test.go

```
@@ -0,0 +1,188 @@
+	router.GET("/", func(c *gin.Context) { c.String(http.StatusOK, "index") })
+	router.GET("/assets/app.js", func(c *gin.Context) { c.String(http.StatusOK, "asset") })
+	router.GET("/api/options", func(c *gin.Context) { c.JSON(http.StatusOK, gin.H{"ok": true}) })
+	router.GET("/ws/trace", func(c *gin.Context) { c.String(http.StatusOK, "ws") })
+	router.Any("/mcp", func(c *gin.Context) { c.String(http.StatusOK, "mcp") })
+	return router
+}
+
+func TestDeployAuthDisabledAllowsRequests(t *testing.T) {
+	router := newDeployAuthTestRouter(deployAuth{})
+	resp := httptest.NewRecorder()
+	req := httptest.NewRequest(http.MethodGet, "/api/options", nil)
```

golangci-lint (2.11.4)

[error] net/http/httptest.NewRequest must not be called. use net/http/httptest.NewRequestWithContext

(noctx)